### PR TITLE
Added spec for create and edit Pool

### PR DESCRIPTION
**Pull request by cloudbehl**

### CODE REVIEW

File: specs/create_Edit_pool.adoc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+== Problem description
+
+For Creating the new Pool we need to Provide forms with different fields
+* Pool Name
+* Number of Pools to create
+* Type()
+* Replicas
+* Cluster it belongs
+* Journal configration
+* OSD's
```
Comment (cloudbehl) [inline specs/create_Edit_pool.adoc]: Which api will give the available osd's value

File: specs/create_Edit_pool.adoc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+* PG Count
```
Comment (cloudbehl) [inline specs/create_Edit_pool.adoc]: Two options are available here.
Either use system generated values or use pg calculator. System generate will be provided by which api?

File: specs/create_Edit_pool.adoc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+* Owner
+* Quotas
```
Comment (cloudbehl) [inline specs/create_Edit_pool.adoc]: owner and quotas do api provide this value?

**Comment by cloudbehl** [comment]
@gnehapk  @kamleshverma1  @anivargi  @nthomas-redhat  @shtripat  Please review this spec.

File: specs/create_Edit_pool.adoc (hunk 2 above)
Comment (shtripat) [reply specs/create_Edit_pool.adoc]: User needs to use PG Calc tool and enter the value. tendrl is not going to provide an auto-calculated value.

File: specs/create_Edit_pool.adoc (hunk 3 above)
Comment (shtripat) [reply specs/create_Edit_pool.adoc]: Quotas values `quota_max_object` and `quota_max_bytes` need to be provided by end user, if quota enabled is selected.

File: specs/create_Edit_pool.adoc [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+For editing a Pool we can edit:
+* Pool Name
+* Replicas  
+* PG-count
+* Quotas
```
Comment (shtripat) [inline specs/create_Edit_pool.adoc:25]: Quota parameters you mean?

File: specs/create_Edit_pool.adoc [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+== Use Cases
+
+The user should be able to create a new Pool with the UI and also Edit the values of Pool before confirming the pool creation.
+
+== Proposed change
+
+* Create a Form for Creating a new pool.
+* Create a new secondary side nav which will have(https://redhat.invisionapp.com/share/2T9LPMEEB#/screens/209774648)
+  - Basic setting
+  - Advance Configration
```
Comment (shtripat) [inline specs/create_Edit_pool.adoc]: spelling `configuration`

File: specs/create_Edit_pool.adoc [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  - Quotas
+* Create a list view For the created pools(https://redhat.invisionapp.com/share/2T9LPMEEB#/screens/217733678)
+* Create review page with data list view.(https://redhat.invisionapp.com/share/2T9LPMEEB#/screens/217733679)
+* List view should have edit button through which we can edit the flowing item listed above.
```
Comment (shtripat) [inline specs/create_Edit_pool.adoc]: spelling `following`

File: specs/create_Edit_pool.adoc [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+== Dependencies:
+https://github.com/Tendrl/dashboard/issues/131
+
+== Testing:
+
+- If create form matches with UI design
+- Next, cancel and Create Buttons are working correctly
+- List view is created after the creatin of Pools
+- You can edit from the list view by clicking on edit button.
```
Comment (shtripat) [inline specs/create_Edit_pool.adoc]: I remember Ju talking about dialog based editing of pools and rbds. @julienlim please confirm.

File: specs/create_Edit_pool.adoc (hunk 1 above)
Comment (cloudbehl) [reply specs/create_Edit_pool.adoc]: @anivargi can you provide input on this?

File: specs/create_Edit_pool.adoc (hunk 2 above)
Comment (cloudbehl) [reply specs/create_Edit_pool.adoc]: @shtripat so should we show both options in UI or should we remove the system generated one?

File: specs/create_Edit_pool.adoc (hunk 3 above)
Comment (cloudbehl) [reply specs/create_Edit_pool.adoc]: @shtripat Quotas total value is provided on the UI. So I think need it from the API.
You can see the 128GB written in UI. I am talking about that value. https://redhat.invisionapp.com/share/2T9LPMEEB#/screens/217733677

File: specs/create_Edit_pool.adoc (hunk 4 above)
Comment (cloudbehl) [reply specs/create_Edit_pool.adoc:25]: Yes I mean quota_max_object and quota_max_bytes.

File: specs/create_Edit_pool.adoc (hunk 2 above)
Comment (shtripat) [reply specs/create_Edit_pool.adoc]: I dont think we are going to have system generated option at all. @julienlim can you confirm?

File: specs/create_Edit_pool.adoc (hunk 3 above)
Comment (shtripat) [reply specs/create_Edit_pool.adoc]: @cloudbehl that 128 GB is not quota value, rather its total size available. Thats not a property of pool. It should be coming for the whole pool based on total size available on OSD to data storage.

**Pull request merged**